nise/generators/aws/ec2_generator.py

```diff
@@ -72,47 +72,59 @@ class EC2Generator(AWSGenerator):
             "0.133",
             "0.133",
             "0.067",
-            "${} per On Demand Linux {} Instance Hour",
+            "${cost} per On Demand Linux {inst_type} Instance Hour",
         ),
     )
 
     ARCHS = ("32-bit", "64-bit")
 
+    OPERATING_SYSTEMS = (
+        "Amazon Linux",
+        "Ubuntu",
+        "Windows Server",
+        "Red Hat Enterprise Linux",
+        "SUSE Linux Enterprise Server",
+        "openSUSE Leap",
+        "Fedora",
+        "Fedora CoreOS",
+        "Debian",
+        "CentOS",
+        "Gentoo Linux",
+        "Oracle Linux",
+        "FreeBSD",
+    )
+
     def __init__(self, start_date, end_date, currency, payer_account, usage_accounts, attributes=None, tag_cols=None):
         """Initialize the EC2 generator."""
         super().__init__(start_date, end_date, currency, payer_account, usage_accounts, attributes, tag_cols)
         self._processor_arch = choice(self.ARCHS)
         self._resource_id = "i-{}".format(self.fake.ean8())
         self._product_sku = self.fake.pystr(min_chars=12, max_chars=12).upper()
         self._instance_type = choice(self.INSTANCE_TYPES)
-        if self.attributes:
-            if self.attributes.get("processor_arch"):
-                self._processor_arch = self.attributes.get("processor_arch")
-            if self.attributes.get("resource_id"):
-                self._resource_id = "i-{}".format(self.attributes.get("resource_id"))
-            if self.attributes.get("product_sku"):
-                self._product_sku = self.attributes.get("product_sku")
-            if self.attributes.get("tags"):
-                self._tags = self.attributes.get("tags")
-            instance_type = self.attributes.get("instance_type")
-            if instance_type:
-                self._instance_type = (
-                    instance_type.get("inst_type"),
-                    instance_type.get("physical_cores"),
-                    instance_type.get("vcpu"),
-                    instance_type.get("memory"),
-                    instance_type.get("storage"),
-                    instance_type.get("family"),
-                    instance_type.get("cost"),
-                    instance_type.get("rate"),
-                    instance_type.get("saving"),
-                    "${} per On Demand Linux {} Instance Hour",
-                )
+        self._operating_system = choice(self.OPERATING_SYSTEMS)
+        self._processor_arch = self.attributes.get("processor_arch")
+        self._resource_id = "i-{}".format(self.attributes.get("resource_id"))
+        self._product_sku = self.attributes.get("product_sku")
+        self._tags = self.attributes.get("tags")
```

Comment:
Specify default values instead of `None`.
```suggestion
        self._operating_system = choice(self.OPERATING_SYSTEMS)
        self._processor_arch = self.attributes.get("processor_arch", choice(self.ARCHS))
        self._resource_id = f"i-{self.attributes.get("resource_id", self.fake.ean8())}"
        self._product_sku = self.attributes.get("product_sku", self.fake.pystr(min_chars=12, max_chars=12).upper())
        self._tags = self.attributes.get("tags", [])
```